### ISSUE TITLE
Cannot run debug gdb with multiple folders in a workspace

### ISSUE THREAD

**Issue by doubleell** [numaru/vscode-ceedling-test-adapter#98]
Hello,
I have followed steps to run Ceedling tests using VS Code. The tests run ok, however when I click the debug icon I have the following messages;
![image](https://user-images.githubusercontent.com/36256378/162189070-e2ebe262-c05e-4b0c-9b74-f7ad81a085de.png)

I have setup the debug launch etc, I have used in the past no problem and pointed to ceedling_gdb.

        {
            "name": "ceedling_gdb",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/ceedling/build/test/out/${command:ceedlingExplorer.debugTestExecutable}",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": true,
            "MIMode": "gdb",
            "miDebuggerPath": "C:/msys64/mingw32/bin/gdb.exe",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ]
        },

**Comment by numaru** [comment]
Hello, thank you for the report. A new release has been published this weekend but nothing should have changed in the "debug part". I tried to debug in the wsl today and did not encounter any issue.

Can you reproduce it with the v1.8.2?

![image](https://user-images.githubusercontent.com/23616600/162281530-9a0e6b14-6784-48de-958f-928c5e297f64.png)

**Comment by doubleell** [comment]
Hi,
Thanks for the reply. I tried v1.8.2 but that was the same. I can’t see
what the problem is, strange….

**Comment by doubleell** [comment]
Hi,
I have just found the issue. I had another folder open in the workspace
that doesn't have any ceedling yaml file. I think it was somehow trying to
run tests for this project instead even though I had clicked the icon for
the correct debug, the run icon for the test worked ok;

[image: image.png]

Regards,
Allan.

**Comment by numaru** [comment]
Sure! This need be to be investigated

**Comment by FTchips** [comment]
Hello,
I am also running into this issue. I try to debug a test by hitting the debug icon in the test explorer, and nothing happens and the same three exceptions pop up.
I had fixed it by specifying the folder in the [workspaceFolder variable](https://code.visualstudio.com/docs/editor/variables-reference#_variables-scoped-per-workspace-folder) in the debug launch: 
```
{
    "name": "Ceedling Test Explorer Debug",
    "type": "cppdbg",
    "request": "launch",
    "program": "${workspaceFolder:**MyFolder**}/build/test/out/${command:ceedlingExplorer.debugTestExecutable}",
    "args": [],
    "stopAtEntry": false,
    "cwd": "${workspaceFolder:**MyFolder**}",
    "environment": [],
    "externalConsole": false,
    "MIMode": "gdb",
    "miDebuggerPath": "C:\\Ruby27-x64\\msys64\\mingw64\\bin\\gdb.exe",
    "setupCommands": [
        {
            "description": "Enable pretty-printing for gdb",
            "text": "-enable-pretty-printing",
            "ignoreFailures": true
        }
    ]
}
```
This worked and I was able to debug tests in all folders. 

However, after adding another folder to the workspace, it seems to have broken again. I have added the debug launch to the new folder with  correct scope, and added the launch name to the debugConfiguration, added the project.yml but still encounter the errors. 
I can run tests from all folders in the workspace, and see statuses for all folders. 
But, I am unable to debug any tests from any of the folders.

Is there a log that might provide more information about where things are going wrong?

**Comment by numaru** [comment]
It should be fixed after #99

**Comment by numaru** [comment]
> Is there a log that might provide more information about where things are going wrong?

Nop, the extension is lacking of this kind of feature. Currently I use the debugger to follow the execution path :smile: 

Thank you both for the report! Feel free to reopen if you still encounter the issue
